app/controllers/home_controller.rb

```diff
@@ -38,9 +40,10 @@ def static
 
   def unsubscribe
     code = params[:code]
-    User.all.select {|x| x.encrypted_email == code }.each {|x| x.update_attributes subscribed_to_newsletter: false }
-    flash[:message] = "You have been unsubscribed from the newsletter."
-    redirect_to "/anime"
+    User.all.select { |x| x.encrypted_email == code }
+      .each { |x| x.update_attributes subscribed_to_newsletter: false }
```

Comment:
Please tell me I'm reading this wrong and it doesn't _actually_ load all users into RAM just to unsubscribe a single one of them?

M-maybe to avoid timing attacks... against our unsubscribe endpoint?  I just... wtf